graphein/molecule/graphs.py

```diff
@@ -227,3 +233,37 @@ def construct_graph(
         g = annotate_edge_metadata(g, config.edge_metadata_functions)
 
     return g
+
+def construct_junction_tree(
+    smiles: Optional[str] = None,
```

Comment:
I think using an RDKit mol as the input makes this function more general (this way we can call it for mols parsed from SDF, PDB, Mol2 etc).

Reply:
I also don't think the input should be optional